storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -3109,10 +3109,1760 @@ buf_block_t *fsp_get_latched_xdes_page(
+  @param prev_block previous block in the list
+  @param next_block next block in the list
+  @return error code */
+  dberr_t remove_prepare(const buf_block_t &descr_block,
+                         uint32_t xoffset, uint32_t free_limit,
+                         mtr_t *mtr, buf_block_t **prev_block,
+                         buf_block_t **next_block) noexcept
+  {
+    const xdes_t *descr= descr_block.page.frame + xoffset;
+    fil_addr_t next_addr= flst_get_next_addr(descr);
+    fil_addr_t prev_addr= flst_get_prev_addr(descr);
+    dberr_t err= DB_SUCCESS;
+
+    if (prev_addr.page != FIL_NULL)
+    {
+      if (!node_valid(&prev_addr))
+        return DB_CORRUPTION;
+
+      *prev_block= fsp_get_latched_page(page_id_t{0, prev_addr.page},
+                                        mtr, &err);
+      ut_ad(!*prev_block == (err != DB_SUCCESS));
+
+      if (!*prev_block)
+        return err;
+
+      fil_addr_t cur_addr=
+        flst_get_next_addr((*prev_block)->page.frame
+                             + prev_addr.boffset);
+      if (cur_addr.page != descr_block.page.id().page_no()
+          || cur_addr.boffset != xoffset)
+        return DB_CORRUPTION;
+    }
+
+    if (next_addr.page != FIL_NULL)
+    {
+      if (!node_valid(&next_addr))
+        return DB_CORRUPTION;
+
+      *next_block= fsp_get_latched_page(page_id_t{0, next_addr.page},
+                                        mtr, &err);
+      ut_ad(!*next_block == (err != DB_SUCCESS));
+      if (!*next_block)
+        return err;
+
+      fil_addr_t cur_addr=
+        flst_get_prev_addr((*next_block)->page.frame
+                             + next_addr.boffset);
+      if (cur_addr.page != descr_block.page.id().page_no()
+          || cur_addr.boffset != xoffset)
+        return DB_CORRUPTION;
+    }
+
+    return err;
+  }
+
+  /** Complete the steps for removing the file list node
+  @param base       base block where free list starts
+  @param boffset    offset where list starts
+  @param descr      descriptor to be removed
+  @param mtr        mini-transaction */
+  void remove_complete(buf_block_t *base, uint16_t boffset,
+                       xdes_t *descr, mtr_t *mtr) noexcept
+  {
+    fil_addr_t next_addr= flst_get_next_addr(descr + XDES_FLST_NODE);
+    fil_addr_t prev_addr= flst_get_prev_addr(descr + XDES_FLST_NODE);
+    /* remove_prepare() checked these already */
+    ut_ad(next_addr.page == FIL_NULL || node_valid(&next_addr));
+    ut_ad(prev_addr.page == FIL_NULL || node_valid(&prev_addr));
+    byte *list= base->page.frame + boffset;
+
+    buf_block_t *prev_block= nullptr;
+    buf_block_t *next_block= nullptr;
+
+    if (prev_addr.page != FIL_NULL)
+    {
+      prev_block=
+        mtr->get_already_latched(page_id_t{0, prev_addr.page},
+                                 MTR_MEMO_PAGE_SX_FIX);
+      ut_ad(prev_block);
+ 
+      flst_write_addr(*prev_block, prev_block->page.frame
+                                     + prev_addr.boffset + FLST_NEXT,
```

Comment:
The second line is indented way too deep (it should be aligned with the `(`) and the line should be split after a binary operator such as `+`, not before them.